### PR TITLE
adds setupDone, unlisten(), camelCased feature flags

### PR DESCRIPTION
**Pull request by rueckstiess**
@imlucas can you please review?

The change includes an `unlisten()` function for each of the metrics, and a `setupDone` boolean. I'm not sure I "unlistened" everything correctly that is set up in `listen()`.

The reason why we want this now is because we can install a listener to `sync` events on `app.preferences` that calls `metrics.listen(app)`. This way, we can toggle any of the tracking on the fly with e.g.

```
app.preferences.save('bugsnag', false)
```

In the end, we can just attach UI toggles to these properties in a preferences pane.

I also camelCased the feature `google-analytics` to `googleAnalytics` for consistency (see `./src/models/preferences.js`). Nicer to work with as these correspond to properties.

### CODE REVIEW

**Comment by rueckstiess** [comment]
network_features

**Comment by kangas** [comment]
@rueckstiess, there is an `exports.listen(app)` in the top-level index.js. Should we add a corresponding unlisten()?

File: lib/google-analytics.js [hunk 1]
```diff
@@ -122,17 +127,41 @@ exports.track = function(title, data) {
...
+    debug('google analytics is disabled');
+    return;
+  }
+
   if (app.user) {
     debug('adding listener to user');
-    app.user.on('sync', exports.setUser.bind(null, app.user));
+    // if user already fetched, set user now
+    if (app.user.fetched) {
+      exports.setUser(app.user);
```
Comment (kangas) [inline lib/google-analytics.js]: You are passing "app.user" as an argument to setUser, but setUser no longer takes an argument above. Do you still want it to take an argument?

File: lib/google-analytics.js [hunk 2]
```diff
@@ -3,28 +3,33 @@ var debug = require('debug')('mongodb-js-metrics:google-analytics');
...
  * @todo (imlucas) Add support for Application Name, ID, Version, etc.
  * https://github.com/peaksandpies/universal-analytics/blob/master/AcceptableParams.md#application-name
  */
 
+exports.visitor = null;
+exports.app = null;
+exports.setupDone = false;
+
 /**
  * @param {Object} [user]
```
Comment (kangas) [inline lib/google-analytics.js]: Remove `@param` tag if the function no longer takes a param

File: lib/intercom.js [hunk 3]
```diff
@@ -184,7 +210,13 @@ exports.listen = function(app) {
 
   if (app.user) {
     debug('adding listener to user to boot intercom');
-    app.user.on('sync', exports.setUser.bind(null, app.user));
+    // if user already fetched, set user now
+    if (app.user.fetched) {
+      exports.setUser(app.user);
```
Comment (kangas) [inline lib/intercom.js]: Passing app.user as an argument, but you removed the "user" argument above.

**Comment by kangas** [comment]
Minor nits found, otherwise LGTM.

**Comment by imlucas** [comment]
published in v0.2.3

**Pull request merged**